### PR TITLE
refactor(connector,core): add config param to email/sms connector test API

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Add `config` param to email/sms connector test API, in order to allow testing the email/sms connector on the fly, without having to save the configs first.

## Testing

- [x] Update email connector config without saving, send test. Received email with new config
- [x] Update SMS connector config without saving, send test. Received SMS on mobile with new config

### CODE REVIEW

**Review by wangsijie — COMMENTED** [review]
`SenderTest` is used in `ConnectorDetails/index.tsx`, do we need to update the usage?

**Comment by demonzoo** [comment]
> `SenderTest` is used in `ConnectorDetails/index.tsx`, do we need to update the usage?

I will, in a follow up PR. Preparing it right now

File: packages/core/src/routes/connector.ts [hunk 1]
```diff
@@ -184,10 +194,11 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
     koaGuard({
       body: object({
         phone: string().regex(phoneRegEx),
+        config: arbitraryObjectGuard.optional(),
       }),
     }),
     async (ctx, next) => {
-      const { phone } = ctx.guard.body;
+      const { phone, config } = ctx.guard.body;
```
[the remark this answers is not in the document]
Comment (demonzoo) [reply packages/core/src/routes/connector.ts:201]: Oh, good catch. I forgot this one

[the remark this answers is not in the document]
Comment (demonzoo) [reply packages/core/src/routes/connector.ts:201]: Updated

**Review by wangsijie — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

**Pull request merged**